### PR TITLE
Bump mapbox gl to 0.44.0

### PR DESCRIPTION
**Pull request by etpinard**
resolves https://github.com/plotly/plotly.js/issues/1714 and makes https://github.com/plotly/plotly.js/pull/2029 obsolete.

It's about time we do this. Note that `mapbox-gl@0.44.0` is **still** incompatible with `nw@0.12` that we use in our imagetest container (ref https://github.com/plotly/plotly.js/pull/2029#issuecomment-345295881), meaning we'll have to _manually_ test mapbox subplot image generation. But yeah after some discussing with @jackparmer, this is way better than sticking to a almost two year old version of `mapbox-gl`.

A lot of things have changed from `mapbox-gl` version `0.22.0` to `0.44.0`. Our tests did catch many of those changes. But still, we'll need to manually test this thing thoroughly before merging.

This PR also adds support for (un)selected marker size and color on top of marker opacity for scattermapbox traces https://github.com/plotly/plotly.js/issues/2346, fixes a hover bug and improves first-render performance.

Some TODOs:

- [x] thorough manual testing
- [x] write more convert test for selected / unselected attributes - done in 1998f90

### CODE REVIEW

**Comment by jackparmer** [comment]
Happy to manually test this tonight or tomorrow on stage!

**Comment by alexcjohnson** [comment]
> meaning we'll have to _manually_ test mapbox subplot image generation.

What do you mean by this? Just that we have tests that can't run on CI but can run locally, or are there things we have to run in a browser and eyeball?

**Comment by etpinard** [comment]
> What do you mean by this? Just that we have tests that can't run on CI but can run locally, or are there things we have to run in a browser and eyeball?

I added a command in [`tasks/noci_test.sh`](https://github.com/plotly/plotly.js/pull/2361/files#diff-ea0c36aa0adf7f9bffb43543231c559bR16) that re-generates the mapbox baselines using [`image-exporter`](https://github.com/plotly/image-exporter). If this doesn't generate a `git diff` we're good. We could use something more evolved at some point, depending how long we'll have to wait before merging https://github.com/plotly/plotly.js/pull/1972.

**Comment by etpinard** [comment]
After some testing help from @jackparmer, tagging this thing as `reviewable`.

File: test/jasmine/tests/scattermapbox_test.js [hunk 1]
```diff
@@ -800,8 +800,8 @@ describe('@noCI Test plotly events on a scattermapbox plot:', function() {
             expect(pt.lon).toEqual(10, 'points[0].lon');
             expect(pt.pointNumber).toEqual(0, 'points[0].pointNumber');
 
-            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
-            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+            expect(evt.clientX).toBeDefined('event.clientX');
+            expect(evt.clientY).toBeDefined('event.clientY');
```
Comment (alexcjohnson) [inline test/jasmine/tests/scattermapbox_test.js]: why did these have to change?

File: test/image/mocks/mapbox_bubbles.json [hunk 2]
```diff
@@ -46,6 +46,29 @@
           "opacity": 0.5
```
Comment (alexcjohnson) [inline test/image/mocks/mapbox_bubbles.json]: Looks (from the baseline image change) like this didn't actually work before, but now it does 🎉 
Was that a known bug?

Comment (etpinard) [reply test/image/mocks/mapbox_bubbles.json]: Yes (after some initial confusion) -> https://github.com/plotly/plotly.js/issues/2346

File: src/traces/scattermapbox/hover.js [hunk 3]
```diff
@@ -10,32 +10,40 @@
...
+        Math.floor((xval + 180) / 360) :
+        Math.ceil((xval - 180) / 360);
 
     // shift longitude to [-180, 180] to determine closest point
     var lonShift = winding * 360;
     var xval2 = xval - lonShift;
 
+    function wrapLon(lon) {
+        return Lib.wrap180(lon);
+    }
```
Comment (alexcjohnson) [inline src/traces/scattermapbox/hover.js]: this wrapper seems unnecessary

File: test/jasmine/tests/scattermapbox_test.js (hunk 1 above)
Comment (etpinard) [reply test/jasmine/tests/scattermapbox_test.js]: Oops. My mistake. I shouldn't have committed this. Good :eye: Reverted in https://github.com/plotly/plotly.js/pull/2361/commits/0ef2ba68623097d39daa1f0cefd6afbc8b064860

File: src/components/drawing/index.js [hunk 4]
```diff
@@ -439,8 +439,8 @@ drawing.pointStyle = function(s, trace, gd) {
     });
 };
 
-drawing.selectedPointStyle = function(s, trace) {
-    if(!s.size() || !trace.selectedpoints) return;
+drawing.makeSelectedPointStyleFns = function(trace) {
```
Comment (alexcjohnson) [inline src/components/drawing/index.js]: Nice abstraction here 👌

**Comment by alexcjohnson** [comment]
@etpinard this looks great! We'll need to sort out fonts before we can use `image_exporter` for everything (just shows up in the legend font in the `mapbox_0` and `mapbox_connectgaps` images, it's a totally different font, look particularly at the c's and g's), but it's OK for this PR. It might mean that you're the only one who can run that `noci_test` script successfully, but if you're OK with that possibility then so am I 😉 

💃 after the one last [minor comment](https://github.com/plotly/plotly.js/pull/2361#discussion_r168021403) and tests pass.

**Comment by etpinard** [comment]
> We'll need to sort out fonts before we can use image_exporter for everything (just shows up in the legend font in the mapbox_0 and mapbox_connectgaps images

Ok. Referencing https://github.com/plotly/plotly.js/pull/1972/

**Pull request merged**